src/main/java/com/netgrif/application/engine/petrinet/service/PetriNetService.java

```diff
@@ -398,14 +396,12 @@ else if (key.equalsIgnoreCase("title") || key.equalsIgnoreCase("initials") || ke
             else
                 valueCriteria = Criteria.where(key).is(value);
             query.addCriteria(valueCriteria);
+            query_total.addCriteria(valueCriteria);
         });
 
         query.with(pageable);
         List<PetriNet> nets = mongoTemplate.find(query, PetriNet.class);
-        return PageableExecutionUtils.getPage(nets.stream()
-                        .map(net -> new PetriNetReference(net, locale)).collect(Collectors.toList()),
-                pageable,
-                () -> mongoTemplate.count(query, PetriNet.class));
+        return new PageImpl<>(nets.stream().map(net -> new PetriNetReference(net, locale)).collect(Collectors.toList()), pageable,  mongoTemplate.count(query_total, PetriNet.class));
```

Comment:
why is the page created like this? The search should not go trough mongoTemplate but trough PetriNetRepository if possible. The repository can do pagination well on its own.

Reply:
It's temporary fix to repair behavior until PetriNetService will be refactored in future. Consulted with @mladoniczky